### PR TITLE
Fix/healthcheck v3

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

- Healthcheck status is set to WARNING if enough brokers are available
- Configurable values of minimum necessary brokers (defaults: 1 for consumers and 2 for producers)

Trello card:  https://trello.com/c/NmmQVJGg/5435-kafka-client-fails-health-check-when-msk-security-patches-are-applied-s1
(this has also been implemented in v2, the only difference is that in v3 the values are configurable)

### How to review

- Make sure code changes make sense (you can compare with existing changes in the latest dp-kafka/v2)
- Make sure unit tests pass

(info) Tested with examples and by using the latest commit of this pr in `dp-cantabular-csv-exporter`.

### Who can review

anyone

### CODE REVIEW

File: health_info.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	}
+
+	if numHealthy >= minHealthyThreshold {
+		if numHealthy == len(*h) {
+			// All brokers are healthy
+			return state.Update(health.StatusOK, msgHealthy, 0)
+		}
+		// Enough brokers are healthy, but not all of them
+		return state.Update(health.StatusWarning, h.ErrorMsg(), 0)
+	}
```
[the remark this answers is not in the document]
Comment (DavidSubiros) [reply health_info.go:44]: thanks! I will do this change.

File: producer_test.go [hunk 2]
```diff
@@ -317,56 +316,56 @@ func TestSend(t *testing.T) {
...
-				p.Checker(ctx, checkState)
-				t1 := time.Now()
-				So(*checkState.LastChecked(), ShouldHappenOnOrBetween, t0, t1)
-				So(checkState.Status(), ShouldEqual, healthcheck.StatusCritical)
-				So(checkState.Message(), ShouldEqual, "unexpected metadata response for broker(s). Invalid brokers: [localhost:9092]")
-			})
-		})
-	})
-}
+// func TestChecker(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (DavidSubiros) [reply producer_test.go:319]: sorry I forgot to remove this commented test

File: health_info.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	// Not enough brokers are healthy
+	return state.Update(health.StatusCritical, h.ErrorMsg(), 0)
+}
+
+// ErrorMsg returns an tailored message according to the information kept in HealthInfoMap
+func (h *HealthInfoMap) ErrorMsg() string {
```
[the remark this answers is not in the document]
Comment (janderson2) [reply health_info.go:64]: Is there a reason this needs to be exported?

[the remark this answers is not in the document]
Comment (DavidSubiros) [reply health_info.go:64]: sorry, I forgot about this change. I think there isn't any reason. I will make it private.

**Review by janderson2 — APPROVED** [review]

**Pull request merged**